src/tools/pr_8799/src/virtcontainers.rs

```diff
@@ -0,0 +1,78 @@
+use anyhow::{anyhow, Result};
+use std::{fs, result::Result::Ok, path::Path, collections::HashMap};
+
+const CPU_POS_STAT: usize = 38;
+const PROC_STAT_SIZE: usize = 52;
+
+// create a map of tids-vcpus from /proc/pid subdirectories
+pub fn get_ch_vcpu_tids_by_path(proc_path: &str) -> Result<HashMap<u32,u32>> {
+
+    let mut tids = HashMap::new();
+
+    if !nix::unistd::Uid::effective().is_root() {
+       return Err(anyhow!("Will not perform pid checks as non root user"));
+    }
+
+    if proc_path.trim().is_empty() {
+        return Err(anyhow!("Path must be non-empty strings."));
+    }
+
+    // canonicalize the /proc/pid path path
+    let c_path = fs::canonicalize(proc_path)?;
+
+    // make the task path
+    let tid_path = fs::canonicalize(Path::new(&c_path).join("task"))?;
+
+    // get the list of tids
+    for entry in fs::read_dir(tid_path)? {
+        let entry = entry?;
+
+        let tid_str = match entry.file_name().into_string() {
+            Ok(id) => id,
+            Err(_) => continue,
+        };
+
+        let tid: u32 = tid_str.parse().expect("Not a valid tid was found");
+
+        // get the cpu_id for each tid
+	// by inspecting: /proc/pid/stat
+        let tid_path = Path::new("/proc").join(tid_str.clone()).join("stat");
+
+	if !tid_path.exists() {
+		return Err(anyhow!("Tid stat was not found."));
+	}
+
+        let proc_pid_stat_line = fs::read_to_string(tid_path)?;
+    
+        let vect_proc_pid_stat: Vec<&str> = proc_pid_stat_line.split(' ').collect();
+
+        // test the size of proc/pid/stat, usually it has a size of 52 fields
+	// See: https://man7.org/linux/man-pages/man5/proc.5.html
+	if vect_proc_pid_stat.len() < PROC_STAT_SIZE {
+            return Err(anyhow!("Status information about the process was not found."));
+        }
+
+        let cpu_id: u32 = vect_proc_pid_stat[CPU_POS_STAT].parse().expect("Not a valid cpu id.");
```

Comment:
Again, can you return the error and not call `expect()` to ensure we don't panic.